src/displayBackend/wayland/Surface.cpp

```diff
@@ -212,6 +212,16 @@ void Surface::init(wl_compositor* compositor)
 
 void Surface::release()
 {
+	{
```

Comment:
If there is a race, I don't see how these changes may help. Changing the order of destroying callback will not solve the issue IMHO.

Reply:
Patch replaced